### PR TITLE
Support settings changes with atomic transactions

### PR DESCRIPTION
**Pull request by GuptaManan100**
## Description

This PR adds the support to store the settings changes that were made in an atomic transaction so that redoing them works with those settings.

Vitess handles `SET <variable>=<value>` commands in different ways based on the settings variable that is being set - 
- `IgnoreThese` - Some variables like `big_tables` are just ignored by Vitess.
- `NotSupported` - Trying to set variables like `binlog_row_image` just fails.
- `ReadOnly` - These variables like `version` can only be read and not updated.
- `CheckAndIgnore` - These variables are ignored when tried to update.
- `VitessAware` - These variables change vitess behaviour. Sometimes, they are stored in the session and also sent down to vttablets to be set on MySQL. 
- `UseReservedConn` - We don't use reserved connections any more (we use the settings pool), but settings here are sent down to vttablet and are applied on the connections. 

Some of the settings can change how some of the DML queries interact. For example, the `time_zone` variable affects what we get from the `NOW()` function call. For settings of `UseReservedConn` type, that we apply on the connection of an atomic transaction, we need to keep track of them in our list of recorded queries, so that when we try to redo the transaction (incase of a failure), we don't end up changing the behaviour of the queries. This PR accomplishes this change. 

## Related Issue(s)

- #16245 

## Checklist

-   [x] "Backport to:" labels have been added if this change should be back-ported to release branches
-   [x] If this change is to be back-ported to previous releases, a justification is included in the PR description
-   [x] Tests were added or are not required
-   [ ] Did the new or modified tests pass consistently locally and on CI?
-   [ ] Documentation was added or is not required

### CODE REVIEW

File: go/vt/vttablet/tabletserver/tx_engine.go [hunk 1]
```diff
@@ -707,12 +723,19 @@ func (te *TxEngine) Release(connID int64) error {
...
+	// If we have a settings query that we need to apply, we do that before starting the transaction.
+	if settingsQuery != "" {
+		if _, err = dbConn.ExecOnce(ctx, settingsQuery, 1, false); err != nil {
+			return nil, err
+		}
+	}
+
 	sc := &StatefulConnection{
 		dbConn: &connpool.PooledConn{
 			Conn: dbConn,
```
Comment (harshit-gangal) [inline go/vt/vttablet/tabletserver/tx_engine.go:741]: Let's add a test case if this is not done this way what can fail.

File: go/test/endtoend/transaction/twopc/stress/stress_test.go [hunk 2]
```diff
@@ -42,6 +42,109 @@ import (
...
+			queries:         append([]string{"begin"}, getMultiShardInsertQueries()...),
+			verifyFunc: func(t *testing.T, vtParams *mysql.ConnParams) {
+				// There is nothing to verify.
+			},
+		},
+		{
+			name:            "Settings changes before begin",
+			commitDelayTime: "5",
+			queries: append(
+				append([]string{`set @@time_zone="+10:30"`, "begin"}, getMultiShardInsertQueries()...),
```
Comment (harshit-gangal) [inline go/test/endtoend/transaction/twopc/stress/stress_test.go:73]: nit: multiple append might not be needed.

Comment (harshit-gangal) [reply go/test/endtoend/transaction/twopc/stress/stress_test.go:73]: this is done multiple places

Comment (GuptaManan100) [reply go/test/endtoend/transaction/twopc/stress/stress_test.go:73]: They're required. getMultiShardInsertQueries is returning a list of strings, so they need to be passed with ellipses. That also means that anything after it, requires a second append call.

Comment (harshit-gangal) [reply go/test/endtoend/transaction/twopc/stress/stress_test.go:73]: I missed there were more statements after that.

**Review by harshit-gangal — APPROVED** [review]

File: go/vt/vttablet/tabletserver/tx_engine.go (hunk 1 above)
Comment (GuptaManan100) [reply go/vt/vttablet/tabletserver/tx_engine.go:741]: Added!

**Review by systay — APPROVED** [review]

**Pull request merged**